presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,315 @@
+import io.prestosql.plugin.hive.metastore.Table;
+import io.prestosql.spi.PrestoException;
+import org.apache.hadoop.hive.metastore.api.Decimal;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_INVALID_METADATA;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createBinaryColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createBooleanColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createDateColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createDecimalColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createDoubleColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createIntegerColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.HiveColumnStatistics.createStringColumnStatistics;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.fromMetastoreDistinctValuesCount;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.fromMetastoreNullsCount;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getAverageColumnLength;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getTotalSizeInBytes;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.toMetastoreDistinctValuesCount;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category.PRIMITIVE;
+
+public class GlueStatConverter
+{
+    private GlueStatConverter() {}
+
+    private static final long DAY_TO_MILLISECOND_FACTOR = TimeUnit.DAYS.toMillis(1);
+
+    public static List<ColumnStatistics> toGlueColumnStatistics(
+            Partition partition, Map<String, HiveColumnStatistics> prestoColumnStats, OptionalLong rowCount)
+    {
+        List<ColumnStatistics> catalogColumnStatisticsList = new ArrayList<>(prestoColumnStats.size());
+
+        prestoColumnStats.forEach((columnName, statistics) -> {
+            final Optional<Column> column = columnByName(partition, columnName);
+            HiveType columnType = column.get().getType();
+
+            ColumnStatistics catalogColumnStatistics = new ColumnStatistics();
+
+            catalogColumnStatistics.setColumnName(columnName);
+            catalogColumnStatistics.setColumnType(columnType.toString());
```

Comment:
Are the various allowed column types documented somewhere? Can there be a case where a HiveType isn't supported by Glue?

Reply:
Same in `toGlueColumnStatistics`.

Reply:
The switch case follow the aws glue api reference, if the type is not supported a `TrinoException` with `HIVE_INVALID_METADATA` is thrown in both `toGlueColumnStatisticsData` and `fromGlueColumnStatistics`,
Do you think we can handle this better ? Should I also add the docs link as reference to the to/from methods ? 

api reference: https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-common.html#aws-glue-api-common-ColumnStatistics

Reply:
This makes sense. Consider this resolved.